ledger/simulation/simulation_eval_test.go

```diff
@@ -1475,6 +1573,570 @@ int 1`
 	})
 }
 
+// The program is copied from pyteal source for c2c test over betanet:
+// source: https://github.com/ahangsu/c2c-testscript/blob/master/c2c_test/max_depth/app.py
```

Comment:
It seems like this program isn't exactly the same as the one you linked?

Reply:
yes, I add the optin and clear out inner txn in second `itxn_submit`.

Reply:
yes, I described the minor difference and pasted modified version of pyteal in gist; [testing teal code explanation](https://github.com/algorand/go-algorand/pull/5360/commits/ef711fc219db72e8b5193895bfd6454e0385e14f)